docs/union.md

```diff
@@ -0,0 +1,150 @@
+# Unions
+
+Consider this Oomph code:
+
+    func foo(Str | null message):
+        if message == null:
+            print("No message")
+        else:
+            print(message.get())
+
+    export func main():
+        foo(null)       # prints "No message"
+        foo("hello")    # prints "hello"
+
+Let's go through this step by step.
+
+    func foo(Str | null message):
+
+Here `Str | null` is a union type. Read it as "`Str` or `null`".
+When the code runs, each instance of the union type has exactly one **active member**,
```

Comment:
I don't think you should use the word 'instances' here. How about 'When the code runs, union type has exactly one **active member**?

Reply:
```
<Arrinao> When the code runs, each instance of the union type can only have one of the given type (in this case Str or null)
<Arrinao> how about that
<Akuli> maybe "be" instead of "have" ?
<Arrinao> okay
```